source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -1,44 +1,116 @@
+
+      **MySQL Mitigation Strategies**
+
+      **Run combined queries prior to the upgrade.**
+      The following queries can be combined when run prior to the upgrade:
+
+      ``ALTER TABLE Posts MODIFY COLUMN FileIds text, MODIFY COLUMN Props JSON;``
+
+      This limits the time taken to that of a single query of that type.
+
+      **Online migration**: An online migration that avoids locking can be attempted on MySQL installations, especially for particularly heavy queries or very big datasets (tens of millions of posts or more). This can be done through an external tool like `pt-online-schema-change <https://www.percona.com/doc/percona-toolkit/LATEST/pt-online-schema-change.html>`__. However, the online migration process can cause a significant spike in CPU usage on the database instance it runs.
+
+      See the `Mattermost v6.0 DB Schema Migrations Analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#online-migration-mysql>`__ documentation for a sample execution and additional caveats.
+
+   .. tab:: PostgreSQL
 
-Upgrading from a previous Extended Support Release to the latest Extended Support Release is supported. However, we strongly recommend that you review the :doc:`important-upgrade-notes` for all intermediate versions in between to ensure you're aware of the possible migrations that could affect your upgrade.
+      The following query executed during the migration process will have a significant impact on database CPU usage and write operation restrictions for the duration of the query:
 
-For `High Availability <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ deployments, we only support one minor version difference between  server versions when performing a rolling upgrade. For example v5.27.1 + v5.27.2 or v5.26.4 + v5.27.1 is supported, whereas v5.25.5 + v5.27.0 is not supported. Running two different versions of Mattermost in your cluster should not be done outside of an upgrade scenario. In some upgrade scenarios, it won't be possible to run different versions, and such cases are clearly noted in the :doc:`important-upgrade-notes` product documentation.
+      ``ALTER TABLE posts ALTER COLUMN props TYPE jsonb USING props::jsonb;`` (~ 11 minutes)
+
+      For a complete breakdown of PostgreSQL queries, as well as their impact and duration, see the `Mattermost v6.0 DB Schema Migrations Analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#postgresql-1>`__.
+
+Upgrading from Releases Older than v5.35
+----------------------------------------
+
+Customers upgrading from a release older than Mattermost v5.35 should expect downtime when upgrading to v6.0, due to the introduction of backend database architecture introduced in v5.35. See the `Mattermost Changelog <https://docs.mattermost.com/install/self-managed-changelog.html?highlight=changelog#id39>`__ documentation for details.
```

Comment:
I'd say they should expect "extended downtime" since on top of 6.0 they will have to run v5.35 changes. Not something I'd recommend on big installations.